server/service/handler.go

```diff
@@ -715,6 +710,7 @@ func attachNewStyleFleetAPIRoutes(r *mux.Router, svc fleet.Service, opts []kitht
 
 	e.GET("/api/v1/fleet/software", listSoftwareEndpoint, listSoftwareRequest{})
 
+	e.GET("/api/v1/fleet/host_summary", getHostSummaryEndpoint, getHostSummaryRequest{})
```

Comment:
Since I had to use the decoder to read request, I might as well move that handler to the new pattern, which is what you see here and below.